src/protocol.rs

```diff
@@ -407,6 +416,132 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    //#[cfg_attr(feature = "flame_it", flame)]
+    #[test]
+    fn process_single_message_wrong_session_id() -> Result<()> {
```

Comment:
todo: For test naming, I like to have the name be almost a sentence describing what property is being tested. This makes it way easier when you run the tests to understand what is actually failing (because `cargo test` just prints out the long list of test names with no further context). In this case, I might do something like
```rust
fn participant_rejects_messages_with_wrong_session_id()
```
or maybe
```rust
fn process_single_message_rejects_mismatched_session_id()
```